src/log-pcap.c

```diff
@@ -213,15 +240,58 @@ static int PcapLogCloseFile(ThreadVars *t, PcapLogData *pl)
     if (pl != NULL) {
         PCAPLOG_PROFILE_START;
 
-        if (pl->pcap_dumper != NULL)
+        if (pl->pcap_dumper != NULL) {
             pcap_dump_close(pl->pcap_dumper);
+#ifdef HAVE_LIBLZ4
+            PcapLogCompressionData *comp = &pl->compression;
+            if (comp->format == PCAP_LOG_COMPRESSION_FORMAT_LZ4) {
+                /* pcap_dump_close() has closed its output ``file'',
+                 * so the buffer has been freed - allocate it again. */
+                comp->pcap_buf = SCMalloc(comp->pcap_buf_size);
```

Comment:
pcap_dump_close doesn't free this buffer, so this leads to a memory leak that is picked up by ASAN. Perhaps all that is needed is a memset?

Reply:
Ouch, thanks for pointing that out! I thought that the buffer will always be freed if you close a pointer from fmemopen. But it seems that's not the case. It's only freed if you made fmemopen allocate the buffer itself by passing it a NULL pointer.